api/pkg/apis/v1alpha1/managers/targets/targets-manager.go

```diff
@@ -245,3 +267,25 @@ func (t *TargetsManager) GetState(ctx context.Context, id string, namespace stri
 	}
 	return ret, nil
 }
+
+func (t *TargetsManager) ValidateCreateOrUpdate(ctx context.Context, state model.TargetState) error {
+	old, err := t.GetState(ctx, state.ObjectMeta.Name, state.ObjectMeta.Namespace)
+	return validation.ValidateCreateOrUpdateWrapper(ctx, state, old, err)
+}
+
+func (t *TargetsManager) ValidateDelete(ctx context.Context, name string, namespace string) error {
+	state, err := t.GetState(ctx, name, namespace)
+	return validation.ValidateDeleteWrapper(ctx, state, err)
+}
+
+func (t *TargetsManager) targetUniqueNameLookup(ctx context.Context, displayName string, namespace string) (interface{}, error) {
+	return states.GetObjectStateWithUniqueName(ctx, t.StateProvider, validation.Target, displayName, namespace)
+}
+
+func (t *TargetsManager) targetInstanceLookup(ctx context.Context, name string, namespace string) (bool, error) {
+	instanceList, err := states.ListObjectStateWithLabels(ctx, t.StateProvider, validation.Instance, namespace, map[string]string{"target": name}, 1)
```

Comment:
Is this name supporting two formats: target1:v1 or target1-v-v1?

Reply:
No. It only expect resource name. The caller needs to convert reference to resource name if necessary.